sklearn/cluster/tests/test_k_means.py

```diff
@@ -540,22 +579,27 @@ def test_score(algo):
     km2 = KMeans(n_clusters=n_clusters, max_iter=10, random_state=42, n_init=1,
                  algorithm=algo)
     s2 = km2.fit(X).score(X)
-    assert_greater(s2, s1)
+    assert s2 > s1
 
 
+@pytest.mark.parametrize('Estimator', [KMeans, MiniBatchKMeans])
 @pytest.mark.parametrize('data', [X, X_csr], ids=['dense', 'sparse'])
 @pytest.mark.parametrize('init', ['random', 'k-means++', centers.copy()])
-def test_predict_minibatch(data, init):
-    mb_k_means = MiniBatchKMeans(n_clusters=n_clusters, init=init,
-                                 n_init=10, random_state=0).fit(data)
+def test_predict(Estimator, data, init):
+    k_means = Estimator(n_clusters=n_clusters, init=init,
+                        n_init=10, random_state=0).fit(data)
 
     # sanity check: re-predict labeling for training set samples
-    assert_array_equal(mb_k_means.predict(data), mb_k_means.labels_)
+    assert_array_equal(k_means.predict(data), k_means.labels_)
 
     # sanity check: predict centroid labels
-    pred = mb_k_means.predict(mb_k_means.cluster_centers_)
+    pred = k_means.predict(k_means.cluster_centers_)
     assert_array_equal(pred, np.arange(n_clusters))
 
+    # re-predict labels for training set using fit_predict
+    pred = k_means.fit_predict(data)
+    assert_array_equal(pred, k_means.labels_)
```

Comment:
Side note: I guess this consistency could be enforced in common tests in `check_clustering` --  but then we wouldn't indeed tests all init methods as we are doing here.